beep/generate_protocol.py

```diff
@@ -89,96 +108,88 @@ def generate_protocol_files_from_csv(csv_filename, output_directory=None):
+                                                                      waveform_name,
+                                                                      template=template_fullpath)
+                else:
+                    protocol = Procedure.generate_procedure_regcyclev3(index,
+                                                                       protocol_params,
+                                                                       template=template_fullpath)
+                protocol.generate_procedure_diagcyclev3(
+                    protocol_params["capacity_nominal"], diagnostic_params
+                )
             else:
-                protocol = Procedure.generate_procedure_regcyclev3(index,
-                                                                   protocol_params,
-                                                                   template=template_fullpath)
-            protocol.generate_procedure_diagcyclev3(
-                protocol_params["capacity_nominal"], diagnostic_params
-            )
-            filename = "{}.000".format(filename_prefix)
-            filename = os.path.join(output_directory, "procedures", filename)
-        elif template == "drivingV1.000":
-            diag_params_df = pd.read_csv(
-                os.path.join(PROCEDURE_TEMPLATE_DIR, "PreDiag_parameters - DP.csv")
-            )
-            diagnostic_params = diag_params_df[
-                diag_params_df["diagnostic_parameter_set"]
-                == protocol_params["diagnostic_parameter_set"]
-                ].squeeze()
-            mwf_dir = os.path.join(output_directory, "mwf_files")
-            waveform_name = insert_driving_parametersv1(protocol_params,
-                                                        waveform_directory=mwf_dir)
-            template_fullpath = os.path.join(PROCEDURE_TEMPLATE_DIR, template)
-            protocol = Procedure.generate_procedure_drivingv1(index,
-                                                              protocol_params,
-                                                              waveform_name,
-                                                              template=template_fullpath)
-            protocol.generate_procedure_diagcyclev3(
-                protocol_params["capacity_nominal"], diagnostic_params
-            )
+                failure = {
+                    "comment": "Unable to find template: " + template,
+                    "error": "Not Found",
+                }
+                file_generation_failures.append(failure)
+                warnings.warn("Unsupported file template {}, skipping.".format(template))
+                result = "error"
+                continue
+
             filename = "{}.000".format(filename_prefix)
             filename = os.path.join(output_directory, "procedures", filename)
-        elif template == "formationV1.mps":
+
+        elif ".mps" in template and template == "formationV1.mps":
```

Comment:
Some more comments next to these would be helpful too.

Reply:
Yup, adding comments to each of the cases